lib/money/money.rb

```diff
@@ -8,6 +8,16 @@ class Money
 
   class << self
     alias_method :from_amount, :new
+
+    def new(value = 0, _currency = nil)
+      if value == 0
+        @empty ||= from_amount(0).freeze
+      else
+        from_amount(value, _currency)
+      end
+    end
```

Comment:
I find it kind of weird that the 0 value is immutable, but other values aren't. When are we actually mutating these?